lib/Parser/ptree.cpp

```diff
@@ -4,6 +4,20 @@
 //-------------------------------------------------------------------------------------------------------
 #include "ParserPch.h"
 
+void ParseNode::Init(OpCode nop, charcount_t ichMin, charcount_t ichLim)
```

Comment:
Dumb question, and maybe @boingoing knows the answer- why not just make this a constructor instead of an explicitly named Init function?

Reply:
and then use placement new at the call sites? Is that what you mean?

Reply:
and then use placement new at the call sites? Is that what you mean?

---
In reply to: [173610992](https://github.com/Microsoft/ChakraCore/pull/4804#discussion_r173610992) [](ancestors = 173610992)

Reply:
the placement new is wrapped by AllocatorNew

---
In reply to: [173858893](https://github.com/Microsoft/ChakraCore/pull/4804#discussion_r173858893) [](ancestors = 173858893,173610992)

Reply:
the inits could be replaced by constructors + AllocatorNew at the callsite, however the constructors have to be propagated through the inheritance hierarchy and would hide default ones. I think we could wait with this till ParseNode* classes get more flesh and then we should consistently provide all of them with constructors (and remove Init methods if any).

---
In reply to: [173897962](https://github.com/Microsoft/ChakraCore/pull/4804#discussion_r173897962) [](ancestors = 173897962,173858893,173610992)